stdlib/public/SDK/Foundation/Foundation.swift

```diff
@@ -1256,12 +1256,39 @@ extension NSDictionary {
   ///   found in `otherDictionary`.
   @objc(_swiftInitWithDictionary_NSDictionary:)
   public convenience init(dictionary otherDictionary: NSDictionary) {
-    // FIXME: This is a bit weird. Maybe there's a better way?
-    self.init(dictionary: otherDictionary as [NSObject: AnyObject]
-                                          as [AnyHashable: Any])
+    let numElems = otherDictionary.count
+    let stride = MemoryLayout<AnyObject>.stride
+    let alignment = MemoryLayout<AnyObject>.alignment
+    let singleSize = stride * numElems
+    let totalSize = singleSize * 2
+    _sanityCheck(stride == MemoryLayout<NSCopying>.stride)
+    _sanityCheck(alignment == MemoryLayout<NSCopying>.alignment)
+
+    // Allocate a buffer containing both the keys and values.
+    let buffer = UnsafeMutableRawPointer.allocate(bytes: totalSize, alignedTo: alignment)
+    defer {
+      buffer.deallocate(bytes: totalSize, alignedTo: alignment)
+      _fixLifetime(otherDictionary)
```

Comment:
Without the fixLifetime, the _stdlib_NSDictionary_getObjects call would be the last use of otherDictionary and it could be released including the values/keys.
At least I think that this is the case - I didn't try without fixLifetime.
So fixLifetime ensures that the values/keys stored in the buffer are kept alive until the final NSDictionary.init call.